src/app/questionnaire-item/store/transform-initial-state.ts

```diff
@@ -350,3 +331,20 @@ export const transformQuestionnaire = (
     },
   },
 });
+
+export const getInitActions = (path: string[]) => (item: any): Action[] => [
+  ...(Array.isArray(item?.answer)
+    ? [
+        setAnswers(
+          [...path, item.linkId],
+          R.map(getExistingValueProperty, item.answer)
```

Comment:
This effectively the only changed line after our second discussion (`getExistingValueProperty` tests for all the `valueX` properties and reads their value). This is the same function which is used for the initial values so the behaviour should be the same as there.